spec/ddtrace/profiling/ext/cthread_spec.rb

```diff
@@ -20,6 +20,11 @@
       expect(::Thread.ancestors).to_not include(described_class)
 
       klass = ::Thread.dup
+
+      # keep copy of original initialize method, useful when we want to simulate a thread where our extension hasn't
+      # been initialized (e.g. a thread created before the extension was added)
+      klass.send(:alias_method, :original_initialize, :initialize)
```

Comment:
Interesting, I like this idea of using `alias_method` for keeping `initialize`.

Reply:
As I touched on above, it was quite hard to simulate this "I add another module after the cthread module, and it breaks cthread initialization" (which is good news, I guess). In particular, if you create the alias AFTER including cthread, the alias will include the cthread initialize.